### DISCUSSION TITLE
Related to execution/es6

### DISCUSSION THREAD

**Discussion by 87sam12** [webpack/webpack#15180] [Q&A]
# Bug report

**What is the current behavior?**

![image](https://user-images.githubusercontent.com/13358855/149645488-c515f9e1-140c-4843-aea5-d657797b33ab.png)
![image](https://user-images.githubusercontent.com/13358855/149645544-bb32c50e-f7bd-4670-8a32-c01ee4bf8703.png)

Even after the control reaches the let assignment statement, the previous imports are undefined. Could anyone explain why is that so?
Also the following shows that the bundled internal variables are defined though:
![image](https://user-images.githubusercontent.com/13358855/149645560-e0595dfe-bdfc-494d-8a07-f373a2a7095a.png)
![image](https://user-images.githubusercontent.com/13358855/149645562-10f8ed38-7309-48a9-ac18-31c8430e3ff1.png)
Why does this behavior exist in javascript? Is this expected or something different.
Its in mozilla firefox browser (95.0.1)
Its undefined even for npm libraries as well, which is in node_modules folder. Why does this happen, which seems slightly not so logical. Is there any issue with the bundler?

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Create a react app. Introduce redux in it.
2. Import that store and provide it in the index.js while rendering the root component.

**What is the expected behavior?**
Just pls explain why is this happening in the browser (firefox 95.0.1) ? Because of this none of the variables imported have any valid values. How to use them in the file post that lines' execution?
Pl give your answer in a more descriptive manner instead of just saying that this behavior is expected. What is exactly this behavior. If this exists how can anyone use any of the imports in the file below? Whats the purpose of importing something you need above, if it will never get imported even after that line is executed?

 As far as I know as javascript is a synchronous language except the specific functionalities like promises, async/await, I guess everything else must be synchronous by its design, which is why this unpredictability in its behavior is confusing.

**Other relevant information:**
webpack version:  5.66.0
Node.js version: 16.13.2
Operating System: Linux
Additional tools:

**Comment by alexander-akait** [comment]
Please create reproducible test repo

**Reply by 87sam12** [reply]
Please find it here: 
https://github.com/12sam12/dummy_app
